### ISSUE TITLE
auto-scaling group with multiple instance types and purchase options (asg + ec2-fleet)

### ISSUE THREAD

**Issue by erks** [terraform-providers/terraform-provider-aws#6478]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

ASG and EC2-fleet are now working together. See https://aws.amazon.com/blogs/aws/new-ec2-auto-scaling-groups-with-multiple-instance-types-purchase-options/. I believe more attributes will have to be added to `aws_autoscaling_group`.

### New or Affected Resource(s)

* aws_autoscaling_group
* aws_ec2_fleet

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* https://aws.amazon.com/blogs/aws/new-ec2-auto-scaling-groups-with-multiple-instance-types-purchase-options/

**Comment by bflad** [comment]
Support for AutoScaling Mixed Instances Policy was released with [version 1.44.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1440-november-14-2018) earlier this week. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by erks** [comment]
Missed that. Thanks @bflad.
